### PR TITLE
[prop-types] Update type definition

### PR DESCRIPTION
**Pull request by retyui**
- Links to documentation: https://www.npmjs.com/package/prop-types
- Link to GitHub or NPM: https://www.npmjs.com/package/prop-types
- Type of contribution: refactor

### CODE REVIEW

**Comment by retyui** [comment]
via: https://github.com/flow-typed/flow-typed/issues/3265

File: definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/prop-types_v15.x.x.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  propName: string,
+  componentName: string,
+  href?: string
+) => ?Error;
+
+// Copied from: https://github.com/facebook/flow/blob/0938da8d7293d0077fbe95c3a3e0eebadb57b012/lib/react.js#L433-L449
+declare module 'prop-types' {
+  declare var array: React$PropType$Primitive<Array<any>>;
+  declare var bool: React$PropType$Primitive<boolean>;
+  declare var func: React$PropType$Primitive<Function>;
```
Comment (pascalduez) [inline definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/prop-types_v15.x.x.js]: I know it's copied from the React sources, but `Function`, `Object` is any.

Comment (retyui) [reply definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/prop-types_v15.x.x.js]: I know too) 
What do you suggest I need to change?

File: definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/test_prop-types.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  it('should passes when used properly', () => {
+    PropTypes.checkPropTypes(
+      { foo: PropTypes.string },
+      { foo: 'foo' },
+      'value',
+      'TestComponent'
+    );
+  });
+
+  it('rises an error when do not pass all required arguments', () => {
```
Comment (pascalduez) [inline definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/test_prop-types.js]: Nit: `s/rise/raise/` ;)

File: definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/prop-types_v15.x.x.js (hunk 1 above)
Comment (pascalduez) [reply definitions/npm/prop-types_v15.x.x/flow_v0.89.x-/prop-types_v15.x.x.js]: Always the same dilemma... No perfect solutions I'm afraid. `() => void` or `() => {}` for `Function`, `{ +[string]: mixed }` for `Object`...

**Pull request merged**